rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -175,11 +176,18 @@ def is_key_in_yaml(cls, file_path: Union[Text, Path], *keys: Text) -> bool:
               `True` if all the keys are contained in the file, `False` otherwise.
 
         Raises:
-            YamlException: if the file seems to be a YAML file (extension) but
-                can not be read / parsed.
+            FileNotFoundException: if the file cannot be found.
         """
-        content = rasa.shared.utils.io.read_yaml_file(file_path)
-        return any(key in content for key in keys)
+        try:
+            with open(file_path) as file:
+                return any(
+                    all(line.startswith(f"{key}:") for key in keys) for line in file
```

Comment:
You're right — it would not be found in this case, but I would prefer to keep it that way because this is how it worked before. This function is a part of `rasa.shared` and it means it's better to not change the behaviour.

However, I updated the docstring.

Reply:
ok I see - it's top level keys only.